backend/src/main/java/ca/bc/gov/app/service/ches/ChesCommonServicesService.java

```diff
@@ -0,0 +1,150 @@
+        ChesMailEncoding.UTF_8,
+        "FSA_donotreply@gov.bc.ca",
+        ChesMailPriority.NORMAL,
+        "Forest Client Application Confirmation",
+        null,
+        requestContent.emailTo()
+    );
+
+    return
+        webClient
+            .post()
+            .uri(configuration.getUri())
+            .header(HttpHeaders.AUTHORIZATION, "Bearer " + getToken())
+            .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+            .body(Mono.just(request), ChesMailRequest.class)
+            .retrieve()
+            .onStatus(httpStatusCode -> httpStatusCode.value() == 401,
+                response -> Mono.error(new InvalidAccessTokenException()))
+            .onStatus(httpStatusCode -> httpStatusCode.value() == 403,
+                response -> Mono.error(new InvalidRoleException()))
+
+            .onStatus(httpStatusCode -> httpStatusCode.value() == 400,get400ErrorMessage())
+            .onStatus(httpStatusCode -> httpStatusCode.value() == 422,get422ErrorMessage())
+            .onStatus(HttpStatusCode::isError, get500ErrorMessage())
+            .bodyToMono(ChesMailResponse.class)
+
+            .map(response -> response.txId().toString())
+            .doOnError(error -> log.error("Failed to send email",error));
+
+  }
+
+  private static Function<ClientResponse, Mono<? extends Throwable>> get500ErrorMessage() {
+    return response ->
+        response
+            .bodyToMono(ChesMailErrorResponse.class)
+            .flatMap(errorMessageDetail -> Mono.error(
+                new UnexpectedErrorException(errorMessageDetail.status(),
+                    errorMessageDetail.detail())));
+  }
+
+  private static Function<ClientResponse, Mono<? extends Throwable>> get422ErrorMessage() {
+    return response ->
+        response
+            .bodyToMono(ChesMailErrorResponse.class)
+            .map(details ->
+                Optional
+                    .ofNullable(details.errors())
+                    .map(errorList -> errorList
+                        .stream()
+                        .map(errorEntry -> String.format("%s on %s", errorEntry.message(),
+                            errorEntry.value()))
+                        .collect(Collectors.joining(","))
+                    )
+                    .map(errors -> String.join(",", errors, details.detail()))
+                    .orElse(details.detail())
+            )
+            .flatMap(errorMessageDetail -> Mono.error(
+                new UnableToProcessRequestException(errorMessageDetail)));
+  }
+
+  private static Function<ClientResponse, Mono<? extends Throwable>> get400ErrorMessage() {
+    return response ->
+        response
+            .bodyToMono(ChesMailErrorResponse.class)
+            .doOnNext(System.out::println)
+            .map(ChesMailErrorResponse::detail)
+            .flatMap(errorMessageDetail -> Mono.error(
+                new BadRequestException(errorMessageDetail)));
+  }
+
+  private String getToken() {
+
```

Comment:
can we use spring boot provided methods than third party libraries as it will handle token life cylce et..
which would also allow us to customize logging and other features at a single place
something like below.

 `@Bean
  @Autowired
  WebClient webClient(final WebClient.Builder builder) {
    val clientRegistryRepo = new InMemoryReactiveClientRegistrationRepository(ClientRegistration
      .withRegistrationId(this.props.getClientID())
      .tokenUri(this.props.getTokenURL())
      .clientId(this.props.getClientID())
      .clientSecret(this.props.getClientSecret())
      .authorizationGrantType(AuthorizationGrantType.CLIENT_CREDENTIALS)
      .build());
    val clientService = new InMemoryReactiveOAuth2AuthorizedClientService(clientRegistryRepo);
    val authorizedClientManager =
      new AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager(clientRegistryRepo, clientService);
    val oauthFilter = new ServerOAuth2AuthorizedClientExchangeFilterFunction(authorizedClientManager);
    oauthFilter.setDefaultClientRegistrationId(this.props.getClientID());
    return builder
      .filter(oauthFilter)
      .build();
  }
`

Reply:
We definitely can. This is a great suggestion, but I will discuss it first with Maria and Catherine to understand why they have chosen that specific library. I will come back to you once I find out and we can discuss what you've proposed.